### PR TITLE
Fix undefined references for set_zero_all

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: include/deal.II/lac/la_vector.h [hunk 1]
```diff
@@ -160,6 +160,15 @@ namespace LinearAlgebra
...
            const bool omit_zeroing_entries = false) override;
 
+    /**
+     * Returns `false` as this is a serial block vector.
+     *
+     * This functionality only needs to be called if using MPI based vectors and
+     * exists in other objects for compatibility.
+     */
+    bool
+    has_ghost_elements() const;
```
Comment (bangerth) [inline include/deal.II/lac/la_vector.h]: Is this from a different part of your work? It doesn't match the PR subject.

Comment (masterleinad) [reply include/deal.II/lac/la_vector.h]: No, we need that for being able to instantiate the function. It seems that this is the only vector class that is missing that function.

File: include/deal.II/lac/la_vector.h [hunk 2]
```diff
@@ -160,6 +160,15 @@ namespace LinearAlgebra
     reinit(const VectorSpaceVector<Number> &V,
            const bool omit_zeroing_entries = false) override;
 
+    /**
+     * Returns `false` as this is a serial block vector.
```
Comment (drwells) [inline include/deal.II/lac/la_vector.h]:
```suggestion
     * Returns `false` as this is a serial vector.
```

File: doc/news/changes/minor/20190905DanielArndt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Fixed: Implementing LinearAlgebra::has_ghost_elements() improves compatibility
```
Comment (kronbichler) [inline doc/news/changes/minor/20190905DanielArndt]:
```suggestion
Fixed: Implementing LinearAlgebra::Vector::has_ghost_elements() improves compatibility
```

**Comment by masterleinad** [comment]
We should merge this soonish so the testers work again.

**Comment by kronbichler** [comment]
> We should merge this soonish so the testers work again.

I agree. If nobody objects within the next two hours, I will merge.

**Comment by drwells** [comment]
This patch seems completely uncontroversial so lets merge.

**Pull request merged**